components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/dao/DAOConstants.java

```diff
@@ -139,102 +139,103 @@ private SQLQueries() {
+                        "FROM IDN_XACML_POLICY WHERE POLICY_ID = :POLICY_ID_1; AND TENANT_ID= :TENANT_ID;) " +
+                        "AND TENANT_ID = :TENANT_ID_1;";
+        public static final String GET_PAP_POLICY_REFS_SQL = "SELECT REFERENCE FROM IDN_XACML_POLICY_REFERENCE " +
+                "WHERE POLICY_ID=:POLICY_ID; AND VERSION=:VERSION; AND TENANT_ID=:TENANT_ID;";
         public static final String GET_PAP_POLICY_SET_REFS_SQL =
-                "SELECT SET_REFERENCE FROM IDN_XACML_POLICY_SET_REFERENCE WHERE TENANT_ID=:TENANT_ID; AND " +
-                        "POLICY_ID=:POLICY_ID; AND VERSION=:VERSION;";
+                "SELECT SET_REFERENCE FROM IDN_XACML_POLICY_SET_REFERENCE WHERE " +
+                        "POLICY_ID=:POLICY_ID; AND VERSION=:VERSION; AND TENANT_ID=:TENANT_ID;";
         public static final String GET_PAP_POLICY_EDITOR_DATA_SQL =
                 "SELECT DATA_ORDER, DATA FROM IDN_XACML_POLICY_EDITOR_DATA WHERE POLICY_ID=:POLICY_ID; AND " +
                         "VERSION=:VERSION; AND TENANT_ID=:TENANT_ID;";
         public static final String GET_PAP_POLICY_META_DATA_SQL = "SELECT ATTRIBUTE_ID, ATTRIBUTE_VALUE, DATA_TYPE, " +
-                "CATEGORY FROM IDN_XACML_POLICY_ATTRIBUTE WHERE POLICY_ID=:POLICY_ID; " +
-                "AND VERSION=:VERSION; AND TENANT_ID=:TENANT_ID;";
+                "CATEGORY FROM IDN_XACML_POLICY_ATTRIBUTE WHERE POLICY_ID=:POLICY_ID; AND VERSION=:VERSION; " +
+                "AND TENANT_ID=:TENANT_ID;";
         public static final String GET_PAP_POLICY_BY_VERSION_SQL = "SELECT POLICY_ID, LAST_MODIFIED_TIME, " +
-                "LAST_MODIFIED_USER, IS_ACTIVE, POLICY_ORDER, POLICY_TYPE, POLICY_EDITOR, POLICY, TENANT_ID, VERSION " +
-                "FROM IDN_XACML_POLICY WHERE IS_IN_PAP = :IS_IN_PAP; AND TENANT_ID = :TENANT_ID; AND " +
-                "POLICY_ID = :POLICY_ID; AND VERSION = :VERSION;";
+                "LAST_MODIFIED_USER, IS_ACTIVE, POLICY_ORDER, POLICY_TYPE, POLICY_EDITOR, POLICY, VERSION, TENANT_ID " +
+                "FROM IDN_XACML_POLICY WHERE IS_IN_PAP = :IS_IN_PAP; AND POLICY_ID = :POLICY_ID; AND " +
+                "VERSION = :VERSION; AND TENANT_ID = :TENANT_ID;";
         public static final String GET_ALL_PAP_POLICIES_SQL = "SELECT t1.POLICY_ID, t1.VERSION, t1" +
                 ".LAST_MODIFIED_TIME, t1.LAST_MODIFIED_USER, t1.IS_ACTIVE, t1.POLICY_ORDER, t1.POLICY_TYPE, " +
                 "t1.POLICY_EDITOR, t1.POLICY, t1.TENANT_ID FROM IDN_XACML_POLICY t1 WHERE t1.IS_IN_PAP = :IS_IN_PAP; " +
-                "AND t1.TENANT_ID = :TENANT_ID; AND t1.VERSION =(SELECT MAX(VERSION) FROM IDN_XACML_POLICY t2 WHERE " +
-                "t2.POLICY_ID = t1.POLICY_ID AND t2.TENANT_ID=:TENANT_ID_1;)";
+                "AND t1.VERSION =(SELECT MAX(VERSION) FROM IDN_XACML_POLICY t2 WHERE " +
+                "t2.POLICY_ID = t1.POLICY_ID AND t2.TENANT_ID = :TENANT_ID;) AND t1.TENANT_ID = :TENANT_ID_1;";
```

Comment:
Any issue using `TENANT_ID` itself, rather defining `TENANT_ID_1`?

Reply:
With our implementation of `NamedPreparedStatements`, it replaces the first occurence of `TENANT_ID` with the given value, therefore setting the value twice with the same name will only set the value for the first occurence.

Reply:
Ack.

Reply:
This could treat as a bug in the NamedPreparedStatements.. shall we open an issue for this?

Reply:
@Yoshani checked further on this and noted this can be fixed.
Fixed in https://github.com/wso2/carbon-identity-framework/pull/5835
(Specifically with https://github.com/wso2/carbon-identity-framework/compare/436f8d85611f994bcb21b7aa9632b1c64977a86d..0e006e9a0cbde615a7ee340e5e3d2480219beee7)